examples/vae/train_vae.py

```diff
@@ -0,0 +1,150 @@
+mnist = data.load_mnist_data()
+mnist['data'] = mnist['data'].astype(np.float32)
+mnist['data'] /= 255
+mnist['target'] = mnist['target'].astype(np.int32)
+
+N = 60000
+x_train, x_test = np.split(mnist['data'],   [N])
+y_train, y_test = np.split(mnist['target'], [N])
+N_test = y_test.size
+
+# Prepare VAE model, defined in net.py
+model = net.VAE(784, n_latent, 500)
+if args.gpu >= 0:
+    cuda.get_device(args.gpu).use()
+    model.to_gpu()
+xp = np if args.gpu < 0 else cuda.cupy
+
+# Setup optimizer
+optimizer = optimizers.Adam()
+optimizer.setup(model)
+
+# Init/Resume
+if args.initmodel:
+    print('Load model from', args.initmodel)
+    serializers.load_npz(args.initmodel, model)
+if args.resume:
+    print('Load optimizer state from', args.resume)
+    serializers.load_npz(args.resume, optimizer)
+
+# Learning loop
+for epoch in six.moves.range(1, n_epoch + 1):
+    print('epoch', epoch)
+
+    # training
+    perm = np.random.permutation(N)
+    sum_loss = 0       # total loss
+    sum_rec_loss = 0   # reconstruction loss
+    for i in six.moves.range(0, N, batchsize):
+        x = chainer.Variable(xp.asarray(x_train[perm[i:i + batchsize]]))
+        optimizer.update(model.get_loss_func(), x)
+        if epoch == 1 and i == 0:
+            with open('graph.dot', 'w') as o:
+                g = computational_graph.build_computational_graph(
+                    (model.loss, ), remove_split=True)
+                o.write(g.dump())
+            print('graph generated')
+
+        sum_loss += float(model.loss.data) * len(x.data)
+        sum_rec_loss += float(model.rec_loss.data) * len(x.data)
+
+    print('train mean loss={}, mean reconstruction loss={}'
+          .format(sum_loss / N, sum_rec_loss / N))
+
+    # evaluation
+    sum_loss = 0
+    sum_rec_loss = 0
+    for i in six.moves.range(0, N_test, batchsize):
+        x = chainer.Variable(xp.asarray(x_test[i:i + batchsize]),
+                             volatile='on')
+        loss_func = model.get_loss_func(k=10, train=False)
+        loss_func(x)
+        sum_loss += float(model.loss.data) * len(x.data)
+        sum_rec_loss += float(model.rec_loss.data) * len(x.data)
+        del model.loss
+    print('test  mean loss={}, mean reconstruction loss={}'
+          .format(sum_loss / N_test, sum_rec_loss / N_test))
+
+
+# Save the model and the optimizer
+print('save the model')
+serializers.save_npz('mlp.model', model)
+print('save the optimizer')
+serializers.save_npz('mlp.state', optimizer)
+
+
+# show results
+if args.gpu >= 0:
+    model.to_cpu()
+
+
+# original images and reconstructed images
+def save_images(x, filename):
+    fig, ax = plt.subplots(3, 3, figsize=(9, 9), dpi=100)
+    for ai, xi in zip(ax.flatten(), x):
+        ai.imshow(xi.reshape(28, 28))
+    fig.savefig(filename)
+
+train_ind = [1, 3, 5, 10, 2, 0, 13, 15, 17]
+x = chainer.Variable(np.asarray(x_train[train_ind]), volatile='on')
+x1 = model(x)
+save_images(x.data, 'train')
+save_images(x1.data, 'train_ae')
```

Comment:
I think `train_reconstructed` is more preferable as the output filename, although it is the matter of tastes.